### PR TITLE
Consolidate more methods in Frame

### PR DESCRIPTION
**Pull request by vyasr**
This PR standardizes some more methods that were previously duplicated between different Frame classes into Frame. Many of the changes should be straightforward to review, with the main possible exception being the changes to serialization logic. Note that because the format of the serialized data has been standardized across different Frame types, objects serialized with old versions of cudf will be incompatible with the new format. I've included some backwards compatibility layers to be removed in a future release. Additionally, this PR adds some methods to Frame that are really only applicable to DataFrame and Series. Since some such methods already exist in Frame, I think it's fine to continue adding to those pending the eventual creation of a common IndexedFrame class.

### CODE REVIEW

File: python/cudf/cudf/core/frame.py [hunk 1]
```diff
@@ -1729,34 +1901,26 @@ def round(self, decimals=0, how="half_even"):
...
+            decimals = {name: decimals for name in self._column_names}
+        elif not isinstance(decimals, abc.Mapping):
             raise TypeError(
                 "decimals must be an integer, a dict-like or a Series"
             )
 
+        cols = {
+            name: col.round(decimals[name], how=how)
+            # TODO: Is this the expected behavior for decimal dtypes, that
+            # rounding to a lower number does not reduce precision?
```
Comment (shwina) [inline python/cudf/cudf/core/frame.py:1918]: Yes -- reducing precision is a different op (casting).

**Review by shwina — APPROVED** [review]
This looks really good to me. Should we signal this PR as breaking? (my inclination is that we shouldn't)

File: python/cudf/cudf/core/frame.py [hunk 2]
```diff
@@ -1729,34 +1901,26 @@ def round(self, decimals=0, how="half_even"):
...
+            if (name in decimals and _is_non_decimal_numeric_dtype(col.dtype))
```
Comment (vyasr) [inline python/cudf/cudf/core/frame.py:1919]:
```suggestion
            if (name in decimals and _is_non_decimal_numeric_dtype(col.dtype))
```

**Comment by vyasr** [comment]
> This looks really good to me. Should we signal this PR as breaking? (my inclination is that we shouldn't)

I agree. I think that providing a couple of intermediate releases where we have backwards compatibility for the old serialized format is good but I would only mark API changes as breaking. @galipremsagar may have thoughts as well since he requested that sort of backwards compat for the first MultiIndex refactor (you can still see that code in this PR).

**Review by galipremsagar — APPROVED** [review]

**Comment by galipremsagar** [comment]
IMO looks like we shouldn't have to mark it as breaking for this PR.

**Comment by vyasr** [comment]
rerun tests

**Comment by vyasr** [comment]
@gpucibot merge

**Pull request merged**